src/bittybuzz/src/BittyBuzzUserFunctions.cpp

```diff
@@ -13,22 +13,52 @@ void foreachHostFCallback(bbzheap_idx_t key, bbzheap_idx_t value, void* params)
     bbzobj_t* keyObj = bbzheap_obj_at(key); // NOLINT
     bbzobj_t* valueObj = bbzheap_obj_at(value); // NOLINT
 
-    if (!bbztype_isint(*keyObj) || context->m_err) {
+    if (!bbztype_isint(*keyObj) || keyObj->i.value > (int16_t)context->m_arguments.size() ||
+        keyObj->i.value < 0 || context->m_err) {
         context->m_err = true;
         return;
     }
 
-    if (bbztype_isint(*valueObj)) {
-        context->m_arguments[context->m_length++] =
+    switch (bbztype(*valueObj)) {
+    case BBZTYPE_INT:
+        context->m_arguments[(uint8_t)keyObj->i.value] =
             FunctionCallArgumentDTO((int64_t)valueObj->i.value);
-    } else if (bbztype_isfloat(*valueObj)) {
-        context->m_arguments[context->m_length++] =
+        context->m_length++;
+        break;
+    case BBZTYPE_FLOAT:
+        context->m_arguments[(uint8_t)keyObj->i.value] =
             FunctionCallArgumentDTO(bbzfloat_tofloat(valueObj->f.value));
-    } else {
+        context->m_length++;
+        break;
+    default:
         context->m_err = true;
     }
 }
 
```

Comment:
Not sure about the ForEach in the struct name. and the function. The for each feels misleading since no iteration is made inside the type (which is impossible in the first place) nor the function.

Normally I am not picky for names in PRs but this one bugs me...

Reply:
Rename to ArgNameAndType and added a comment